packages/e2e-tests-playwright/config/test.ts

```diff
@@ -0,0 +1,127 @@
+
+// Register role selector.
+selectors.register( 'role', selectorScript, { contentScript: true } );
+
+let globalPage: Page | null = null;
+
+/**
+ * Set of console logging types observed to protect against unexpected yet
+ * handled (i.e. not catastrophic) errors or warnings. Each key corresponds
+ * to the Playwright ConsoleMessage type, its value the corresponding function
+ * on the console global object.
+ */
+const OBSERVED_CONSOLE_MESSAGE_TYPES = [ 'warn', 'error' ] as const;
+
+/**
+ * Adds a page event handler to emit uncaught exception to process if one of
+ * the observed console logging types is encountered.
+ *
+ * @param  message The console message.
+ */
+function observeConsoleLogging( message: ConsoleMessage ) {
+	const type = message.type();
+	if (
+		! OBSERVED_CONSOLE_MESSAGE_TYPES.includes(
+			type as typeof OBSERVED_CONSOLE_MESSAGE_TYPES[ number ]
+		)
+	) {
+		return;
+	}
+
+	const text = message.text();
+
+	// An exception is made for _blanket_ deprecation warnings: Those
+	// which log regardless of whether a deprecated feature is in use.
+	if ( text.includes( 'This is a global warning' ) ) {
+		return;
+	}
+
+	// A chrome advisory warning about SameSite cookies is informational
+	// about future changes, tracked separately for improvement in core.
+	//
+	// See: https://core.trac.wordpress.org/ticket/37000
+	// See: https://www.chromestatus.com/feature/5088147346030592
+	// See: https://www.chromestatus.com/feature/5633521622188032
+	if ( text.includes( 'A cookie associated with a cross-site resource' ) ) {
+		return;
+	}
+
+	// Viewing posts on the front end can result in this error, which
+	// has nothing to do with Gutenberg.
+	if ( text.includes( 'net::ERR_UNKNOWN_URL_SCHEME' ) ) {
+		return;
+	}
+
+	// TODO: Not implemented yet.
+	// Network errors are ignored only if we are intentionally testing
+	// offline mode.
+	// if (
+	// 	text.includes( 'net::ERR_INTERNET_DISCONNECTED' ) &&
+	// 	isOfflineMode()
+	// ) {
+	// 	return;
+	// }
+
+	// As of WordPress 5.3.2 in Chrome 79, navigating to the block editor
+	// (Posts > Add New) will display a console warning about
+	// non - unique IDs.
+	// See: https://core.trac.wordpress.org/ticket/23165
+	if ( text.includes( 'elements with non-unique id #_wpnonce' ) ) {
+		return;
+	}
+
+	// Ignore all JQMIGRATE (jQuery migrate) deprecation warnings.
+	if ( text.includes( 'JQMIGRATE' ) ) {
+		return;
+	}
+
+	const logFunction = type as typeof OBSERVED_CONSOLE_MESSAGE_TYPES[ number ];
+
+	// Disable reason: We intentionally bubble up the console message
+	// which, unless the test explicitly anticipates the logging via
+	// @wordpress/jest-console matchers, will cause the intended test
+	// failure.
+
+	// eslint-disable-next-line no-console
+	console[ logFunction ]( text );
+}
+
+const test = base.extend< {
+	testUtils: TestUtils;
+} >( {
+	page: async ( { browser }, use ) => {
+		if ( ! globalPage || globalPage.isClosed() ) {
```

Comment:
We'd strongly discourage doing that and recommend context isolation by default. Doing this is probably the cause that video/context/screenshot on failure/on-first-retry is not behaving as expected. Also this ends up in that one test can cause other tests to fail/missbehave.

Background: By default we recommend a browser context per test. A browser context is cheap to create (just a few milliseconds) since it lives in memory. This is one of the big powers when you compare Playwright against alternative testing frameworks.

For longer testing scenarios there are test.steps: https://playwright.dev/docs/api/class-test#test-step

Reply:
For historical reasons, many tests depend on the server state so we can't spawn multiple pages simultaneously for now (I would love to have an option to opt-in to that by test case though 🤔 ). Most of our tests also expect the page to be created in `beforeAll`, overriding the fixture seems like the simplest way to support that. However, using independent context for each test is a great suggestion, and I just pushed a commit to do exactly that 👍 .

Reply:
I tried a different approach after chatting with @mxschmitt. It now instead uses the default `page` and `context` in tests so that video capturing and a bunch of other magics Playwright is doing internally should continue to work.

This also means that `page` and `testUtils` can't be used inside `beforeAll` or `afterAll` though. The solution I proposed here is to use the new fixtures `globalPage` and `globalTestUtils` (names TBD) I created, which internally spawn a new page in `beforeAll`/`afterAll` so that they can still run the utils. Use them sparingly though as some of them should be done in `beforeEach` or `afterEach` instead.